genesis/process/disabled/epochNotifier.go

```diff
@@ -0,0 +1,33 @@
+package disabled
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+)
+
+// EpochNotifier represents a disabled epoch notifier implementation
+type EpochNotifier struct {
+}
+
+// CheckEpoch does nothing
+func (en *EpochNotifier) CheckEpoch(_ uint32) {
+}
+
+// RegisterNotifyHandler will call the non nil handler with the 0 epoch value
+func (en *EpochNotifier) RegisterNotifyHandler(handler core.EpochNotifiedHandler) {
+	if check.IfNil(handler) {
+		return
+	}
+
+	handler.EpochConfirmed(0)
```

Comment:
Yes, there is a special use case for the genesis processing in which we need the SC processor and the tx processor. Due to the fact that the genesis processing is done before the construction of those 2 processors used in the normal operation of the node, we instantiated some temporary processors just to get the genesis job done. So, those 2 processors are allowed from the 0-th epoch to do SC deployment, relayed txs (not required by the genesis block) and builtin functions calls (also not required by the genesis process). In order to "activate" those features, the disabled epoch notifier needs to notify all those processors in order to make them work. Otherwise, the SC deployment would fail in the genesis processing leading to errors being escalated and premature node closing.

Reply:
Indeed there is a special case for the aforementioned use case when it executes the hardfork process. Will fix that.

Reply:
fixed

Reply:
ok